fetcher/fetcher.go

```diff
@@ -85,13 +85,25 @@ func New(
 	serverAddress string,
 	options ...Option,
 ) *Fetcher {
+	// Override transport idle connection settings
+	//
+	// See this conversation around why `.Clone()` is used here:
+	// https://github.com/golang/go/issues/26013
+	defaultTransport := http.DefaultTransport.(*http.Transport).Clone()
+	defaultTransport.IdleConnTimeout = DefaultIdleConnTimeout
+	defaultTransport.MaxIdleConns = DefaultMaxConnections
+	defaultTransport.MaxIdleConnsPerHost = DefaultMaxConnections
```

Comment:
The one behavioral change here for clients who are not passing their own rosetta API client is that we're not leveraging the client specified.

We could potentially change this to occur after applying the `options` and check if a client is initialized, e.g.

```
for _, opt := range options {
	opt(f)
}

if f.rosettaClient == nil {
    f.rosettaClient = newDefaultAPIClient(c.maxConnections)
}
```

Also, that way we're also not building out that client, if it is already being passed in with the `WithClient` option.

Reply:
Great catch! And great suggestion! There is no need for us to init the fetcher with a client at all. We can delay ituntil after the opts and that gives us much more sensible code!